examples/C++/DDS/BasicConfigurationExample/HelloWorldSubscriber.cpp

```diff
@@ -0,0 +1,240 @@
+#include <csignal>
+
+using namespace eprosima::fastdds::dds;
+using namespace eprosima::fastdds::rtps;
+
+std::atomic<bool> HelloWorldSubscriber::stop_(false);
+std::mutex HelloWorldSubscriber::terminate_cv_mtx_;
+std::condition_variable HelloWorldSubscriber::terminate_cv_;
+
+HelloWorldSubscriber::HelloWorldSubscriber()
+    : participant_(nullptr)
+    , subscriber_(nullptr)
+    , topic_(nullptr)
+    , reader_(nullptr)
+    , type_(new HelloWorldPubSubType())
+{
+}
+
+bool HelloWorldSubscriber::is_stopped()
+{
+    return stop_;
+}
+
+void HelloWorldSubscriber::stop()
+{
+    stop_ = true;
+    terminate_cv_.notify_one();
+}
+
+bool HelloWorldSubscriber::init(
+        const std::string& topic_name,
+        uint32_t max_messages,
+        uint32_t domain,
+        const std::string& transport,
+        bool reliable,
+        bool transient)
+{
+    DomainParticipantQos pqos;
+    pqos.name("Participant_sub");
+    if (!transport.empty())
+    {
+        pqos.transport().use_builtin_transports = false;
+        if (transport == "shm")
+        {
+            auto shm_transport = std::make_shared<SharedMemTransportDescriptor>();
+            pqos.transport().user_transports.push_back(shm_transport);
+        }
+        else
+        {
+            auto udp_transport = std::make_shared<UDPv4TransportDescriptor>();
+            pqos.transport().user_transports.push_back(udp_transport);
+        }
+    }
+    participant_ = DomainParticipantFactory::get_instance()->create_participant(domain, pqos);
+
+    if (participant_ == nullptr)
+    {
+        return false;
+    }
+
+    //REGISTER THE TYPE
+    type_.register_type(participant_);
+
+    //CREATE THE SUBSCRIBER
+    subscriber_ = participant_->create_subscriber(SUBSCRIBER_QOS_DEFAULT, nullptr);
+
+    if (subscriber_ == nullptr)
+    {
+        return false;
+    }
+
+    //CREATE THE TOPIC
+    topic_ = participant_->create_topic(
+        topic_name,
+        "HelloWorld",
+        TOPIC_QOS_DEFAULT);
+
+    if (topic_ == nullptr)
+    {
+        return false;
+    }
+
+    // CREATE THE READER
+    if (max_messages > 0)
+    {
+        listener_.set_max_messages(max_messages);
+    }
+    DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
+    if (!transport.empty())
+    {
+        rqos.data_sharing().off();
+    }
+    else
+    {
+        rqos.data_sharing().automatic();    // default
+    }
+
+    if (reliable)
+    {
+        rqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
+    }
+    else
+    {
+        rqos.reliability().kind = BEST_EFFORT_RELIABILITY_QOS;  // default
+    }
+
+    if (transient)
+    {
+        rqos.durability().kind = TRANSIENT_LOCAL_DURABILITY_QOS;
+    }
+    else
+    {
+        rqos.durability().kind = VOLATILE_DURABILITY_QOS;   // default
+    }
+
+    reader_ = subscriber_->create_datareader(topic_, rqos, &listener_);
+
+    if (reader_ == nullptr)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+HelloWorldSubscriber::~HelloWorldSubscriber()
```

Comment:
This whole function assumes `participant_` has been created, but it is created in `init` method and not in constructor, so it should check it (also `subscriber_`)

Reply:
:+1: